drivers/timer/realtek_rts5912_rtmr.c

```diff
@@ -0,0 +1,242 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Copyright (c) 2024 Realtek Semiconductor Corporation, SIBG-SD7
+ * Author: Lin Yu-Cheng <lin_yu_cheng@realtek.com>
+ */
+
+#define DT_DRV_COMPAT realtek_rts5912_rtmr
+
+#include <zephyr/init.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/sys_clock.h>
+#include <zephyr/irq.h>
+#include <zephyr/spinlock.h>
+#include <zephyr/drivers/timer/system_timer.h>
+
+#include <reg/reg_rtmr.h>
+#include <reg/reg_system.h>
+
+#define RTS5912_SCCON_REG_BASE ((SYSTEM_Type *)(DT_REG_ADDR(DT_NODELABEL(sccon))))
+
+#define CYCLES_PER_TICK (CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC / CONFIG_SYS_CLOCK_TICKS_PER_SEC)
+
+#define RTMR_REG ((RTOSTMR_Type *)DT_INST_REG_ADDR(0))
```

Comment:
Generally, all drivers should support multiple instances.  If the driver cannot support multiple instances, it should have a build assert to verify only 1 instance is present.
```
BUILD_ASSERT(DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT) == 1,
	     "multiple instances not supported");
```

Reply:
We will add this assert to prevent the multiple instances case.